### PR TITLE
Update CPA and CES CRDs.

### PR DESCRIPTION
**Pull request by archanapholla**
- In CPA, updated region field to an array.
- In CES, added Account namespace field in specs and introduced status field for capturing error from CES CR.
- UT and integration test changes to comply to new CRD format.

This PR doesn't implement functionality required for supporting the newly added fields.

### CODE REVIEW

File: config/crd/bases/crd.cloud.antrea.io_cloudentityselectors.yaml [hunk 1]
```diff
@@ -41,6 +41,9 @@ spec:
               accountName:
                 description: AccountName specifies cloud account in this CloudProvider.
                 type: string
+              accountNamespace:
```
Comment (reachjainrahul) [inline config/crd/bases/crd.cloud.antrea.io_cloudentityselectors.yaml:44]: Please under the crds in helm dir too.. build/charts/nephe/charts/crd

File: pkg/cloudprovider/cloudapi/aws/aws_account_mgmt.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ func setAccountCredentials(client client.Client, credentials interface{}) (inter
 
 	awsConfig := &awsAccountConfig{
 		AwsAccountCredential: *accCred,
-		region:               strings.TrimSpace(awsProviderConfig.Region),
+		region:               strings.TrimSpace(awsProviderConfig.Region[0]),
```
Comment (reachjainrahul) [inline pkg/cloudprovider/cloudapi/aws/aws_account_mgmt.go:47]: Add a comment why we are using index 0 for now

**Review by Anandkumar26 — COMMENTED** [review]
Docs needs to be updated as well.

File: config/crd/bases/crd.cloud.antrea.io_cloudentityselectors.yaml (hunk 1 above)
Comment (archanapholla) [reply config/crd/bases/crd.cloud.antrea.io_cloudentityselectors.yaml:44]: Copy this generated file to build/charts/nephe/charts/crds/cloudentityselectors.yaml?

Comment (reachjainrahul) [reply config/crd/bases/crd.cloud.antrea.io_cloudentityselectors.yaml:44]: No, manually patch it.

Comment (archanapholla) [reply config/crd/bases/crd.cloud.antrea.io_cloudentityselectors.yaml:44]: done

File: pkg/cloudprovider/cloudapi/aws/aws_account_mgmt.go (hunk 2 above)
Comment (archanapholla) [reply pkg/cloudprovider/cloudapi/aws/aws_account_mgmt.go:48]: added

File: docs/getting-started.md [hunk 3]
```diff
@@ -115,7 +115,8 @@ metadata:
   namespace: sample-ns
```
Comment (Anandkumar26) [inline docs/getting-started.md:115]: The agented-vm-guide.md also needs to be updated for CES.

File: apis/crd/v1alpha1/cloudentityselector_types.go [hunk 4]
```diff
@@ -46,20 +46,29 @@ type VirtualMachineSelector struct {
 type CloudEntitySelectorSpec struct {
 	// AccountName specifies cloud account in this CloudProvider.
```
Comment (Anandkumar26) [inline apis/crd/v1alpha1/cloudentityselector_types.go:47]: nit:  AccountName specifies the name of the CloudProviderAccount.

File: apis/crd/v1alpha1/cloudentityselector_types.go [hunk 5]
```diff
@@ -46,20 +46,29 @@ type VirtualMachineSelector struct {
...
+}
+
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:shortName="ces"
 
 // CloudEntitySelector is the Schema for the cloudentityselectors API.
 type CloudEntitySelector struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              CloudEntitySelectorSpec `json:"spec,omitempty"`
```
Comment (Anandkumar26) [inline apis/crd/v1alpha1/cloudentityselector_types.go:62]: nit: Add a new line after metav1.ObjectMeta `json:"metadata,omitempty"

File: apis/crd/v1alpha1/cloudprovideraccount_types.go [hunk 6]
```diff
@@ -38,14 +38,14 @@ type CloudProviderAccountAWSConfig struct {
...
-	Region string `json:"region,omitempty"`
+	Region []string `json:"region,omitempty"`
 	// Endpoint URL that overrides the default AWS generated endpoint.
 	Endpoint string `json:"endpoint,omitempty"`
 }
 
 type CloudProviderAccountAzureConfig struct {
 	SecretRef *SecretReference `json:"secretRef,omitempty"`
-	Region    string           `json:"region,omitempty"`
+	Region    []string         `json:"region,omitempty"`
```
Comment (Anandkumar26) [inline apis/crd/v1alpha1/cloudprovideraccount_types.go:48]: I don't think we allow region field to be empty.
So, we have to remove 'omitempty' for the region field.

Comment (archanapholla) [reply apis/crd/v1alpha1/cloudprovideraccount_types.go]: Agree that region is a required field, removed the omitEmpty tag.
However, In webhook there is a check already to make sure region field exists, hence not sure removing the omitEmpty field is really required. I assume even SecretReference is mandatory, yet we have omitEmpty tag for it.

**Review by Anandkumar26 — APPROVED** [review]
/LGTM

**Pull request merged**